src/Microsoft.AspNetCore.Mvc.Testing/MvcWebApplicationBuilder.cs

```diff
@@ -0,0 +1,137 @@
+        }
+
+        /// <summary>
+        /// Configures <see cref="ApplicationPartManager"/> to include the default set
+        /// of <see cref="ApplicationPart"/> provided by <see cref="DefaultAssemblyPartDiscoveryProvider"/>.
+        /// </summary>
+        /// <returns>An instance of this <see cref="MvcWebApplicationBuilder{TStartup}"/></returns>
+        public MvcWebApplicationBuilder<TStartup> UseApplicationAssemblies()
+        {
+            ApplicationAssemblies.AddRange(DefaultAssemblyPartDiscoveryProvider
+                .DiscoverAssemblyParts(typeof(TStartup).Assembly.GetName().Name)
+                .Select(s => ((AssemblyPart)s).Assembly)
+                .ToList());
+
+            return this;
+        }
+
+        /// <summary>
+        /// Configures the application to use a constant culture for testing purposes.
+        /// </summary>
+        /// <returns>An instance of this <see cref="MvcWebApplicationBuilder{TStartup}"/></returns>
+        public MvcWebApplicationBuilder<TStartup> UseCultureReplacer()
+        {
+            ConfigureAfterStartup(s => s.TryAddEnumerable(ServiceDescriptor.Transient<IStartupFilter, CultureReplacerStartupFilter>()));
+            return this;
+        }
+
+        /// <summary>
+        /// Configures the application content root.
+        /// </summary>
+        /// <param name="solutionName">The glob pattern to use for finding the solution.</param>
+        /// <param name="solutionRelativePath">The relative path to the content root from the solution file.</param>
+        /// <returns>An instance of this <see cref="MvcWebApplicationBuilder{TStartup}"/></returns>
+        public MvcWebApplicationBuilder<TStartup> UseSolutionRelativeContentRoot(
+            string solutionRelativePath,
+            string solutionName = "*.sln")
+        {
+            var applicationBasePath = AppContext.BaseDirectory;
+
+            var directoryInfo = new DirectoryInfo(applicationBasePath);
+            do
+            {
+                var solutionPath = Directory.EnumerateFiles(directoryInfo.FullName, solutionName).FirstOrDefault();
+                if (solutionPath != null)
+                {
+                    ContentRoot = Path.GetFullPath(Path.Combine(directoryInfo.FullName, solutionRelativePath));
+                    return this;
+                }
+
+                directoryInfo = directoryInfo.Parent;
+            }
+            while (directoryInfo.Parent != null);
+
+            throw new Exception($"Solution root could not be located using application root {applicationBasePath}.");
+        }
+
+        public TestServer Build()
+        {
+            using (new CultureReplacer())
+            {
+                var builder = new WebHostBuilder()
+                    .UseContentRoot(ContentRoot)
+                    .ConfigureServices(InitializeServices);
+
+                builder.UseStartup<TestStartup<TStartup>>();
+
+                return new TestServer(builder);
```

Comment:
It would be nice to have these verify that the `.deps` file is present (and anything else required). This is the most common issue that our users run into.